### PR TITLE
Displaying cards in hand

### PR DESCRIPTION
**Pull request by frazeman**
- using Card from protobuf with enums for Rank and Suit
- adjusted PlayingCard style, using Card as prop
- added hand of cards to Game
- GameBoard displays PlayingCard for each card in hand

### CODE REVIEW

**Review by daveismith — APPROVED** [review]
Is there a way to sort the cards?

**Comment by frazeman** [comment]
> Is there a way to sort the cards?

Sorting is a good idea! That should probably be part of the Game data and there be a setting for how each user would like them sorted? We could let the user manually sort them further if they like.

**Comment by frazeman** [comment]
I see a further improvement might also be to use the `Hand` from `protobuf` as well.

**Comment by deadwing88** [comment]
missing some semicolons throughout but otherwise LGTM

**Review by deadwing88 — APPROVED** [review]

**Comment by deadwing88** [comment]
also noticed the heart is black, we need some way to differentiate which suits are red and which are black

**Comment by frazeman** [comment]
> missing some semicolons throughout but otherwise LGTM

Haha, so used to Swift and not needing any semicolons. 😆

**Comment by frazeman** [comment]
> also noticed the heart is black, we need some way to differentiate which suits are red and which are black

The heart renders as red for me. Maybe just depends on the emoji set your system uses?

It should eventually be replaced with our own art though anyway.

File: frontend/shoot-client/src/components/Common/PlayingCard.tsx [hunk 1]
```diff
@@ -1,39 +1,79 @@
...
+        case Card.Suit.SPADES:
+            return "♠️";
+        case Card.Suit.CLUBS:
+            return "♣️";
+        case Card.Suit.HEARTS:
+            return "♥️";
+        case Card.Suit.DIAMONDS:
+            return "♦️";
+        default:
+            return "X"
```
Comment (frazeman) [inline frontend/shoot-client/src/components/Common/PlayingCard.tsx:21]:
```suggestion
            return "X";
```

File: frontend/shoot-client/src/components/Common/PlayingCard.tsx [hunk 2]
```diff
@@ -1,39 +1,79 @@
...
+        case Card.Rank.TEN:
+            return "10";
+        case Card.Rank.NINE:
+            return "9";
+        case Card.Rank.EIGHT:
+            return "8";
+        case Card.Rank.SEVEN:
+            return "7";
+        default:
+            return "X"
```
Comment (frazeman) [inline frontend/shoot-client/src/components/Common/PlayingCard.tsx:44]:
```suggestion
            return "X";
```

File: frontend/shoot-client/src/components/Game/Game.context.tsx [hunk 3]
```diff
@@ -1,19 +1,31 @@
 import React, { createContext, useContext, useState } from "react";
 
+import { Card } from "../../proto/shoot_pb";
+
 export interface IGame {
     numPlayers: number;
     sceneView: boolean;
     score: number[];
     tricks: number[];
+    hand: Card[]
```
Comment (frazeman) [inline frontend/shoot-client/src/components/Game/Game.context.tsx:10]:
```suggestion
    hand: Card[];
```

File: frontend/shoot-client/src/components/Game/Game.context.tsx [hunk 4]
```diff
@@ -1,19 +1,31 @@
...
 }
 
 type GameContextType = (IGame | ((param: any) => void))[];
 
+const card1 = new Card()
```
Comment (frazeman) [inline frontend/shoot-client/src/components/Game/Game.context.tsx:15]:
```suggestion
const card1 = new Card();
```

File: frontend/shoot-client/src/components/Game/Game.context.tsx [hunk 5]
```diff
@@ -1,19 +1,31 @@
...
+card1.setRank(Card.Rank.ACE)
```
Comment (frazeman) [inline frontend/shoot-client/src/components/Game/Game.context.tsx:16]:
```suggestion
card1.setRank(Card.Rank.ACE);
```

File: frontend/shoot-client/src/components/Game/Game.context.tsx [hunk 6]
```diff
@@ -1,19 +1,31 @@
...
+card1.setSuit(Card.Suit.SPADES)
```
Comment (frazeman) [inline frontend/shoot-client/src/components/Game/Game.context.tsx:17]:
```suggestion
card1.setSuit(Card.Suit.SPADES);
```

File: frontend/shoot-client/src/components/Game/Game.context.tsx [hunk 7]
```diff
@@ -1,19 +1,31 @@
...
+
+const card2 = new Card()
```
Comment (frazeman) [inline frontend/shoot-client/src/components/Game/Game.context.tsx:19]:
```suggestion
const card2 = new Card();
```

File: frontend/shoot-client/src/components/Game/Game.context.tsx [hunk 8]
```diff
@@ -1,19 +1,31 @@
...
+card2.setRank(Card.Rank.JACK)
```
Comment (frazeman) [inline frontend/shoot-client/src/components/Game/Game.context.tsx:20]:
```suggestion
card2.setRank(Card.Rank.JACK);
```

File: frontend/shoot-client/src/components/Game/Game.context.tsx [hunk 9]
```diff
@@ -1,19 +1,31 @@
...
+card2.setSuit(Card.Suit.HEARTS)
```
Comment (frazeman) [inline frontend/shoot-client/src/components/Game/Game.context.tsx:21]:
```suggestion
card2.setSuit(Card.Suit.HEARTS);
```

**Review by frazeman — COMMENTED** [review]
Fix semicolons

**Pull request merged**